spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -426,7 +426,7 @@ private Object deserializePayload(Object payload, MimeType contentType) {
 	}
 
 	private Object deserializePayload(byte[] bytes, MimeType contentType) {
-		if (TEXT_PLAIN.equals(contentType)) {
+		if (TEXT_PLAIN.equals(contentType) || APPLICATION_JSON.equals(contentType)) {
```

Comment:
This is only necessary because of https://github.com/spring-cloud/spring-cloud-stream/pull/394/files#r54971080, which may not be necessary itself. I believe we can just rely on `text/plain`.